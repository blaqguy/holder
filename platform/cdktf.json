{
  "language": "typescript",
  "app": "npx node dist/apps/platform/main.js",
  "projectId": "67ff3db5-543b-4d0d-97f8-b195012d08a6",
  "sendCrashReports": "false",
  "codeMakerOutput": "./libs/generated/.gen/",
  "terraformProviders": [],
  "terraformModules": [
    {
      "name": "AccountFactoryForTerraform",
      "source": "aws-ia/control_tower_account_factory/aws",
      "version": "1.6.4"
    },
    {
      "name": "asg",
      "source": "terraform-aws-modules/autoscaling/aws",
      "version": "6.7.1"
    },
    {
      "name": "ecs",
      "source": "terraform-aws-modules/ecs/aws",
      "version": "5.2.0"
    }
  ]
}
