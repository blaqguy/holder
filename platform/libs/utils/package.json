{
  "name": "@dragonfly/utils",
  "version": "0.0.1"
}
