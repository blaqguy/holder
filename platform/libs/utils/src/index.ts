export * from './lib/utils';
export * from './lib/constants';
export * from './lib/multiRegionDeployment';
export * from './lib/accounts';
