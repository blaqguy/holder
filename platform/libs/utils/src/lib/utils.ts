export * from './packerUtils';
export * from './helpers';
export * from './constants';
export * from './secrets';
export * from './tagging';
export * from './helpers/types';
