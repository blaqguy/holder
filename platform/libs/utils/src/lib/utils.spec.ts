import { Testing } from 'cdktf';
import { Constants, DfPackerUtils } from './utils';

describe('DfPackerUtils', () => {
  it('return AMI data source given an AMI name', () => {
    const synthedMockStack = Testing.synthScope((mockStack) => {
      DfPackerUtils.getAmiByName(mockStack, Constants.AMIS.UOB_APP_SERVER);
    });
    expect(synthedMockStack).toMatchSnapshot();
  });
});
