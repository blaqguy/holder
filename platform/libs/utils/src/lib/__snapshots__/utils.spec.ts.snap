// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DfPackerUtils return AMI data source given an AMI name 1`] = `
"{
  \\"data\\": {
    \\"aws_ami\\": {
      \\"uob-app-server-Datasource\\": {
        \\"most_recent\\": true,
        \\"name_regex\\": \\"uob-app-server\\"
      }
    }
  }
}"
`;
